src/main/kotlin/org/jitsi/jicofo/xmpp/XmppConfig.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2020 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.xmpp
+
+import org.jitsi.metaconfig.config
+import org.jitsi.config.JitsiConfig.Companion.legacyConfig
+import org.jitsi.config.JitsiConfig.Companion.newConfig
+import org.jitsi.metaconfig.optionalconfig
+import org.jxmpp.jid.DomainBareJid
+import org.jxmpp.jid.impl.JidCreate
+import org.jxmpp.jid.parts.Resourcepart
+import java.time.Duration
+
+class XmppConfig {
+    val serviceConnectionConfig = ServiceConnectionConfig()
+
+    val clientConnectionConfig = ClientConnectionConfig()
+
+    companion object {
+        @JvmField
+        val xmppConfig = XmppConfig()
+    }
+}
+
+class ServiceConnectionConfig {
+    private val enabled: Boolean by config {
+        "jicofo.xmpp.service.enabled".from(newConfig)
+    }
+    fun enabled() = legacyHostname != null || enabled
```

Comment:
nit: I don't think it really matters, but I _think_ we could do:
```
val enabled: Boolean by config {
    "org.jitsi.jicofo.BRIDGE_MUC_XMPP_HOST".from(legacyConfig).convertFrom<String> { it != null }
    "jicofo.xmpp.service.enabled".from(newConfig)
}
```

and then get rid of the need for the separate `enabled()` and `legacyHost` fields.  I don't care much if we change it, though.